src/wasm/superfasthash.ts

```diff
@@ -19,31 +19,30 @@ export function hash(messageLength: u32): u32 {
   let digest = messageLength;
 
   for (let n = messageLength >> 2; n > 0; n--) {
-    digest += load<u16>(index);
-    index += 2;
-    digest ^= digest << 16 ^ load<u16>(index) << 11;
-    index += 2;
+    let value = load<u32>(index);
+    index += 4;
+    digest += value & 0xFFFF;
+    digest ^= digest << 16 ^ (value >> 16) << 11;
     digest += digest >> 11;
   }
 
   switch (messageLength & 3) {
-  case 3:
```

Comment:
I had intended to keep both the JS and the Wasm versions as close to the [original implementation in C](http://www.azillionmonkeys.com/qed/hash.html) in terms of how the code is structured. I understand that it is not merely the reordering of the cases and some additional indentation but also the removal of unnecessary increments of `index`. I had tried the latter a few days ago myself but did not find it to add significant value in terms of performance; on the other hand, I like the fact that the value of `index` always ends up at the length of the message by the end of the function.

Reply:
ok. I reverted this changes